cmake/mcuboot.cmake

```diff
@@ -110,8 +110,18 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
     set(CONFIG_BOOT_SIGNATURE_KEY_FILE ${mcuboot_SIGNATURE_KEY_FILE})
   endif ()
 
-  if (DEFINED mcuboot_CONF_FILE)
-    get_filename_component(mcuboot_CONF_DIR ${mcuboot_CONF_FILE} DIRECTORY)
+  foreach (filepath ${mcuboot_CONF_FILE})
+    file(STRINGS ${filepath} mcuboot_CONFIG_BOOT_SIGNATURE_KEY_FILE
+         REGEX "^CONFIG_BOOT_SIGNATURE_KEY_FILE=")
+    if (mcuboot_CONFIG_BOOT_SIGNATURE_KEY_FILE)
+      get_filename_component(mcuboot_CONF_DIR ${filepath} DIRECTORY)
+      break()
```

Comment:
Should we break here?  I think, we should take key from the last mcuboot_CONF_FILE, because the key could be overwritten.

Reply:
right, it should of course be the last conf file that wins, not the first one.
Fixed.